File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/ResourceAssignments/assignmentcost.rb
module Asposetasksjava
  module AssignmentCost
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        resource_assignments = project.getResourceAssignments().toList()

        # Parse through all the collected resource assignments
        i = 0
        while i < resource_assignments.size()
          ra = resource_assignments.get(i)
          puts "COST: " + ra.get(Rjb::import('com.aspose.tasks.Asn').COST).to_string
          puts "ACWP: " + ra.get(Rjb::import('com.aspose.tasks.Asn').ACWP).to_string
          puts "BCWP: " + ra.get(Rjb::import('com.aspose.tasks.Asn').BCWP).to_string
          puts "BCWS: " + ra.get(Rjb::import('com.aspose.tasks.Asn').BCWS).to_string
          puts "--------------------------------------------------------"
          i +=1
        end    
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithResourceAssignments/AssignmentCost.php
<?php
namespace Aspose\Tasks\WorkingWithResourceAssignments;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Asn as Asn;
class AssignmentCost{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $resource_assignments = $project->getResourceAssignments()->toList();

        # Parse through all the collected resource assignments
        $asn=new Asn();
        $i = 0;
        while ($i < sizeof($resource_assignments)) {
          $ra = $resource_assignments -> get($i);
          print "COST: " . (string)$ra -> get($asn -> COST).PHP_EOL;// . to_string
          print "ACWP: " . (string)$ra -> get($asn -> ACWP).PHP_EOL;// . to_string
          print "BCWP: " . (string)$ra -> get($asn -> BCWP).PHP_EOL;// . to_string
          print "BCWS: " . (string)$ra -> get($asn -> BCWS).PHP_EOL;// . to_string
          print "--------------------------------------------------------".PHP_EOL;
          $i += 1;
        }

    }

}
?>